packages/cmd/src/CommandProcessor.ts

```diff
@@ -721,6 +733,122 @@ export class CommandProcessor {
     }
 
+    /**
+     * Print parameters in place for --dry-run in effect
+     * @private
+     * @param {IHandlerParameters} commandParameters
+     * @returns
+     * @memberof CommandProcessor
+     */
+    private showInputsOnly(response: IHandlerResponseApi, commandParameters: IHandlerParameters) {
+
+        /**
+         * Determine if we should display secure values.  If the ENV variable is set to true,
+         * then we will display the secure values.  If the ENV variable is set to false,
+         * then we will display the non-secure values.
+         *                                                             - GitHub Copilot AI
+         */
+        let showSecure = false;
+
+        // ZOWE_SHOW_SECURE_ARGS
+        const SECRETS_ENV = `${ImperativeConfig.instance.envVariablePrefix}${CommandProcessor.ENV_SHOW_SECURE_SUFFIX}`;
+        const env = (process.env[SECRETS_ENV] || "false").toUpperCase();
+
+        if (env === "TRUE" || env === "1") {
+            showSecure = true;
+        }
+
+        // if config exists and a layer exists
+        let useConfig = false;
+        this.mConfig?.layers.forEach((layer) => {
+            if (layer.exists) {
+                useConfig = true;
+            }
+        });
+
+        /**
+         * Determine if Zowe V2 Config is in effect.  If it is, then we will construct
+         * a Set of secure fields from its API.  If it is not, then we will construct
+         * a Set of secure fields from the `ConnectionPropsForSessCfg` defaults.
+         */
+        const secureInputs: Set<string> =
+            useConfig ?
+                new Set(this.mConfig.api.secure.secureFields().map(v => v.split('.').slice(-1)[0])) :
+                new Set([...LoggerUtils.CENSORED_OPTIONS, ...LoggerUtils.SECURE_PROMPT_OPTIONS]);
+        interface IResolvedArgsResponse {
+            commandValues?: ICommandArguments;
+            profileVersion?: `v1` | `v2`;
+            requiredProfiles?: string[];
+            optionalProfiles?: string[];
+            locations?: string[]
+        }
+
+        /**
+         * Build a list of arguments that will be displayed to the user and note
+         * if any are censored.
+         */
+        const showInputsOnly: IResolvedArgsResponse =
+        {
+            commandValues: {} as ICommandArguments,
+            profileVersion: useConfig ? `v2` : `v1`,
+        };
+
+        let censored = false;
+
+        // only attempt to show the input if it is in the command definition
+        for (let i = 0; i < commandParameters.definition.options?.length; i++) {
+            const name = commandParameters.definition.options[i].name;
+
+            if (commandParameters.arguments[name] != null) {
+
+                if (showSecure || !secureInputs.has(name)) {
+                    showInputsOnly.commandValues[name] = commandParameters.arguments[name];
+                } else {
+                    showInputsOnly.commandValues[name] = LoggerUtils.CENSOR_RESPONSE;
```

Comment:
Another option for a "censor response" would be `(secure value)` which is what the "config list" command uses.

Not sure which value is preferred here, just mentioning this for consideration 🙂 Perhaps we set a precedent of using `****` in log files and `(secure value)` in CLI output?